File: go-ut-bench/datasets/python/python_code_files_self_contained/boundary/boundary_001.py
def safe_divide(numerator: float, denominator: float) -> float:
    if denominator == 0:
        raise ZeroDivisionError("Cannot divide by zero")
    if numerator == 0:
        return 0.0
    result = numerator / denominator
    if result > 1e308:
        return float('inf')
    if result < -1e308:
        return float('-inf')
    return result

File: go-ut-bench/datasets/cpp/cpp_code_files_self_contained/boundary/boundary_001.cpp
#include <stdexcept>
#include <cmath>

double safeDivide(double numerator, double denominator) {
    if (denominator == 0) {
        throw std::runtime_error("Cannot divide by zero");
    }
    if (numerator == 0) {
        return 0.0;
    }
    double result = numerator / denominator;
    return result;
}

int safeSqrt(int value) {
    if (value < 0) {
        throw std::invalid_argument("Cannot compute square root of negative number");
    }
    if (value == 0) {
        return 0;
    }
    return static_cast<int>(std::sqrt(static_cast<double>(value)));
}